katrain/gui/popups.py

```diff
@@ -839,9 +841,25 @@ def on_submit(self):
 
 
 class ReAnalyzeGamePopup(BoxLayout):
-    katrain = ObjectProperty(None)
     popup = ObjectProperty(None)
 
+    def on_checkbox_active(self, checkbox, value):
+        self.start_move.disabled = not value
+        self.end_move.disabled = not value
+
+    def __init__(self, katrain, **kwargs):
+        super().__init__(**kwargs)
+
+        self.katrain = katrain
+        self.checkbox = find_child_by_name(self, "move_range")
+        self.checkbox.bind(active=self.on_checkbox_active)
+
+        self.start_move = find_child_by_name(self, "start_move")
+        self.end_move = find_child_by_name(self, "end_move")
+        self.start_move.disabled = True
+        self.end_move.disabled = True
+        self.start_move.text = str(katrain.game.current_node.depth)
+
     def on_submit(self):
         self.button.trigger_action(duration=0)
```

Comment:
please check the kv for how 'button' is a property here, so you can do the same with other fields

Reply:
Oh, I see, I did not understand what was going on with these properties. I think I'll be able to get rid of the get child by name stuff.

Reply:
It's gone.